acra/configuring-maintaining/installing/building-acrawriter.md

```diff
@@ -0,0 +1,262 @@
+
+#### Installation
+```bash
+sudo apt install node npm
+sudo apt install nodejs-legacy
+sudo npm install -g npm
+cd wrappers/nodejs
+npm install nan
+npm install
+```
+#### Testing
+```
+node test.js
+```
+Output
+```
+work
+```
+#### Examples
+[With Zones](https://github.com/cossacklabs/acra/blob/master/examples/nodejs/examples_with_zone.js) and [without Zones](https://github.com/cossacklabs/acra/blob/master/examples/nodejs/examples_without_zones.js).
+
+### Building AcraWriter for PHP
+
+> Before building AcraWriter for PHP, make sure you've installed [Themis as system library]({{< ref "themis/installation/" >}})!
+
+#### Installation
+
+```bash
+sudo apt install php5 php5-dev
+cd themis
+sudo make phpthemis_install
+# extension=phpthemis.so
+sudo nano /etc/php5/cli/php.ini
+# check that extension is available
+php --ri phpthemis
+cd -
+```
+#### Testing
+```bash
+php acra/wrappers/php/test.php
+```
+Output
+```
+work
+```
+Now use `acra/wrappers/php/acrawriter.php` in your project. In future
+`acrawriter` will contain composer.json and will be added to packagist.
+
+#### Examples
+[With Zones](https://github.com/cossacklabs/acra/blob/master/examples/php/example_with_zone.php) and [without Zones](https://github.com/cossacklabs/acra/blob/master/examples/php/example.php)
+
+### Building AcraWriter for Go
+
+> Before building AcraWriter for Go, make sure you've installed [Themis as system library]({{< ref "themis/installation/" >}})!
+
+#### Installation
+```bash
+sudo apt install golang
+# leave your GOPATH or set some
+export GOPATH=$HOME/work
+go get github.com/cossacklabs/acra/acra-writer
+```   
+
+#### Testing
+```
+go test github.com/cossacklabs/acra/acra-writer
+```
+Output
+```
+ok  	github.com/cossacklabs/acra/acra-writer	0.112s
+```
+
+#### Examples
+[With Zones](https://github.com/cossacklabs/acra/blob/master/examples/golang/src/example_with_zone/example_with_zone.go) and [without Zones](https://github.com/cossacklabs/acra/blob/master/examples/golang/src/example/example.go)
+
+### Building AcraWriter for C++
+
+> Before building AcraWriter for C++, make sure you've installed [Themis as system library]({{< ref "themis/installation/" >}})!
+
+#### Dependencies
+
+Additionally, install [ThemisPP (Themis C++ wrapper) as system library](https://docs.cossacklabs.com/themis/languages/cpp/installation/) from Themis source folder:
+
+```bash
+cd themis
+make themispp_install
+cd -
+```
+
+#### Installation
+
+Grab `acrawriter.hpp` source file and add it to your project. Link `themis` and `crypto` libs.
+
+Example of CMake file:
+
+```bash
+project(my_project)
+set(CMAKE_CXX_STANDARD 14)
+
+//path to acrawriter.hpp file
+set(ACRAWRITER ../../wrappers/cpp)
+include_directories(${ACRAWRITER})
+
+set(
+    SOURCE_FILES
+    ${ACRAWRITER}/acrawriter.hpp
+    main.cpp
+    )
+
+add_executable(my_project ${SOURCE_FILES})
+target_link_libraries(my_project themis crypto)
+
+```
```

Comment:
I think this can be just replaced with a snippet of some C++ code, showcasing AcraWriter API, and deferring AcraWriter integration into the build system to CLion example project. Or just link to the project, like it's done for iOS, if the code is quite involved there.